### DISCUSSION TITLE
@soroban-react library for ReactJS dApps! 🚀

### DISCUSSION THREAD

**Discussion by esteblock** [stellar/sorobanathon#86] [General]
Hello!
I have created the [@soroban-react](https://github.com/esteblock/soroban-react/) npm library. Now consists in 3 packages:

|Package|Version|
|--|--|
|[`@soroban-react/core`](packages/core)                | [![npm version](https://img.shields.io/npm/v/@soroban-react/core/latest.svg)](https://www.npmjs.com/package/@soroban-react/core/v/latest)|
|[`@soroban-react/types`](packages/types)                     | [![npm version](https://img.shields.io/npm/v/@soroban-react/types/latest.svg)](https://www.npmjs.com/package/@soroban-react/types/v/latest)|
|[`@soroban-react/freighter`](packages/core)                     | [![npm version](https://img.shields.io/npm/v/@soroban-react/freighter/latest.svg)](https://www.npmjs.com/package/@soroban-react/freighter/v/latest)|

This library was created based on the [soroban-example-dapp](https://github.com/stellar/soroban-example-dapp/tree/main/wallet) repo  written by @paulbellamy , plus some contributions I made in order to make it react in real time and make it more Solidity dev friendly, inspired in the web3-react library. 

Currently only supports Freighter, but it's made in a way that's very easy to add new wallets (connectors), just as web3-react does with EVM connectors!

I also created an[ example dapp on React JS](https://github.com/esteblock/soroban-react-example-cra) documented, and with it's [vercel app](https://soroban-react-example-cra.vercel.app/) as a way to show how easy is to connect a wallet to a ReactJS dapp! 

Also, on the README file of the [library repo](https://github.com/esteblock/soroban-react) you can find some Documentation.

This is just the first version of the library! Of course a lot of things need to be improved. Please join the project by writing Issues and writing pull requests!

**Comment by esteblock** [comment]
I'm happy to tell you that my Pull Request in the [stellar's soroban-example-dapp repo](https://github.com/stellar/soroban-example-dapp) have been [accepted](https://github.com/stellar/soroban-example-dapp/pull/66), and now it does use the [@soroban-react](https://github.com/esteblock/soroban-react/) library ;)

**Comment by achallagundla** [comment]
Heya @esteblock  ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1400 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

**Comment by achallagundla** [comment]
Heya @esteblock I don't see that you filled out the Typeform? Please do so ASAP to claim your reward. 
Thank You,
Anuhya

**Reply by esteblock** [reply]
Will do it tomorrow AM
Thanks!

**Reply by esteblock** [reply]
Hi @achallagundla , sent you a DM on Discord!
